src/algebra/module/localized_module.lean

```diff
@@ -387,4 +387,122 @@ instance localized_module_is_localized_module :
   { mp := λ eq1, by simpa only [one_smul] using localized_module.mk_eq.mp eq1,
     mpr := λ ⟨c, eq1⟩, localized_module.mk_eq.mpr ⟨c, by simpa only [one_smul] using eq1⟩ } }
 
+section
+
+/--
+If `(M', f : M ⟶ M')` satisfies universal property of localized module, there is a canonical map
+`localized_module S M ⟶ M'`.
+-/
+@[simps]
+noncomputable def from_localized_module [is_localized_module S f] :
```

Comment:
I think you can provide an `is_localized_module.lift` analogous to `is_localization.lift` and specialize to this afterwards?

Or is it too tedious without further API that uses this iso?

Reply:
What is the situation with this comment?

Reply:
I just tried to define `is_localized_module.lift` by
```lean
noncomputable def sec [is_localized_module S f] (m : M') : M × S :=
classical.some $ is_localized_module.surj S f m

lemma sec_spec [is_localized_module S f] (m : M') :
  (sec S f m).2 • m = f (sec S f m).1 :=
classical.some_spec $ is_localized_module.surj S f m

def lift [is_localized_module S f] {g : M →ₗ[R] M''} (hg : ∀ (x : S), is_unit ((algebra_map R (module.End R M'')) x)) :
  M' →ₗ[R] M'' :=
{ to_fun := λ m, (hg $ (sec S f m).2).unit⁻¹ $ g (sec S f m).1,
  map_add' := λ x y,
  begin
    generalize_proofs h1 h2 h3,
    erw [module.End_algebra_map_is_unit_inv_apply_eq_iff, smul_add, ←h2.unit⁻¹.1.map_smul, 
      ←h3.unit⁻¹.1.map_smul, ←linear_map.map_smul, ←linear_map.map_smul],
    sorry,
  end,
  map_smul' := _ }
```
but then I got stuck because a lot of API needs to be proved about `sec` so the end result might not be shorter than the current version.

Reply:
I think we want `lift` in any case, it is a more general construction than the existing one, right? In any case we can wait for another PR.

Reply:
I actually finished `lift` as 
```lean
def lift' {g : M →ₗ[R] M''} (hg : ∀ (x : S), is_unit ((algebra_map R (module.End R M'')) x)) :
  (localized_module S M) →ₗ[R] M''
```
so the general lift is obtained via the isomorphism between `localised_module S M` and `is_localized_module S f`, so still a different approach than `localization` in ring_theory file. Do you think this is acceptable @riccardobrasca

Reply:
If you are able that it makes the diagram commutative and it is unique it is acceptable, it doesn't matter how it is constructed (since you proved that it satisfies the universal property). This should also automatically implies that the isomorphism between `localised_module S M` and `M'` is unique (if it makes the diagram commutative).

Reply:
I have proved the uniqueness and commutativity.